spec/lib/bulk_data/cache_spec.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+RSpec.describe BulkData::Cache do
+  describe ".write" do
+    it "adds an entry that expires in 24 hours" do
+      BulkData::Cache.write("key", "value")
+      expect(BulkData::Cache.cache.fetch("key")).to eq("value")
+
+      travel_to(25.hours.from_now) do
+        expect(BulkData::Cache.cache.fetch("key")).to be_nil
+      end
+    end
+
+    it "sets the current time as the age, which also expires" do
+      freeze_time do
+        BulkData::Cache.write("key", "value")
+        expect(BulkData::Cache.cache.fetch("key:created")).to eq(Time.current)
+      end
+
+      travel_to(25.hours.from_now) do
+        expect(BulkData::Cache.cache.fetch("key:created")).to be_nil
+      end
+    end
+  end
+
+  describe ".fetch" do
+    it "returns a value when one exists" do
+      BulkData::Cache.write("key", "value")
+      expect(BulkData::Cache.fetch("key")).to eq("value")
+    end
+
+    it "raises a NoEntryError when there isn't an entry" do
+      expect { BulkData::Cache.fetch("key") }
+        .to raise_error(BulkData::Cache::NoEntryError)
+    end
+  end
+
+  describe ".wrote_at" do
+    it "returns the time the entry was created" do
+      time = 2.hours.ago.change(usec: 0)
```

Comment:
It is a shame indeed. It's because travel_to does it to the time so we need it to match: https://github.com/rails/rails/blob/8bec77cc0f1fd47677a331a64f68c5918efd2ca9/activesupport/lib/active_support/testing/time_helpers.rb#L145

I could do it is an absolute time but I need it to be within the last 24 hours so it's tricky :-/